### PR TITLE
Adds foreground check to DataLayerAppHelper methods

### PR DESCRIPTION
**Pull request by garanj**
#### WHAT

Adds a foreground check to methods provided by DataLayerAppHelper

#### WHY

Methods in DataLayerAppHelper can launch activities on the remote device. This should only be possible where the calling process is in the foreground.

#### HOW

Checks foreground status of process invoking the method

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: datalayer/core/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<resources>
+    <string name="datalayerapphelper_foreground_error">This method can only be called from the foreground.</string>
```
Comment (yschimke) [inline datalayer/core/src/main/res/values/strings.xml:18]: I'm not sure we want the exception translated. Maybe better being in english in log files.

**Review by yschimke — APPROVED** [review]

File: datalayer/core/src/main/res/values/strings.xml (hunk 1 above)
Comment (garanj) [reply datalayer/core/src/main/res/values/strings.xml:18]: Sure. Shall I move it straight out of the resource?

Comment (garanj) [reply datalayer/core/src/main/res/values/strings.xml]: Fixed.

File: datalayer/phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt [hunk 2]
```diff
@@ -47,6 +48,7 @@ public class PhoneDataLayerAppHelper(
 
     @CheckResult
     override suspend fun startCompanion(node: String): AppHelperResultCode {
+        super.checkIsForegroundOrThrow()
```
Comment (yschimke) [inline datalayer/phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt:51]: why `super.` prefix?

Comment (garanj) [reply datalayer/phone/src/main/java/com/google/android/horologist/datalayer/phone/PhoneDataLayerAppHelper.kt]: Fixed

**Comment by yschimke** [comment]
Landing, doesn't seem blocked.

**Pull request merged**